handbook/engineering/distribution/product/roadmap.md

```diff
@@ -166,6 +165,16 @@ We want to move to a world where all of Sourcegraphs' internal infrastructure is
 - Discussions: none
 - Dependencies: none
 
+### Push site admins to use Docker Compose or Kubernetes for production deployments
+
+Many customers of Sourcegraph today are still running a single-container `sourcegraph/server` deployment in production. We recently began advising all new deployments that this deployment option is _not_ for production use because it has no proper resource isolation and as such when it falls over it is impossible to debug, leading to painstakingly urgent migrations to better deployment types and frustrated/angry customers. We would like to get to a world where all production instances of Sourcegraph are Docker Compose or Kubernetes only.
```

Comment:
Does this include deprioritizing `sourcegraph/server` or excluding features for it? ie https://github.com/sourcegraph/sourcegraph/issues/11473 (not specifically saying we shouldn't include this particular feature, but whether there might be cases in the future where we just say "yeah let's not add X to `sourcegraph/server`" as a result of this roadmap item)

I'm mostly asking because I'm curious about the scope of these roadmap items - if this one doesn't include deprioritizing work, it seems to mostly be a case of communicating difficulties about `sourcegraph/server`, which seems like a small thing compared to some of the other roadmap items that feel more like big goals

Reply:
Yes, exactly de-prioritizing `sourcegraph/server`. This could either be in the form of "it is only for demo deployments" or in the form of "we replace it entirely with Docker Compose and a one-liner install script"

The effort needed by engineering to support it (both by you in https://github.com/sourcegraph/sourcegraph/issues/11473 and others like code-intel team) because it's such an odd deployment model is an impact I will clarify here.